server/routes/adjudicationTabbedParent/scheduleHearing/scheduleHearingPage.ts

```diff
@@ -0,0 +1,190 @@
+        await this.reportedAdjudicationsService.scheduleHearing(
+          adjudicationNumber,
+          hearingDetailsToSave.locationId,
+          formatDate(hearingDetailsToSave.hearingDate),
+          user
+        )
+      }
+      return res.redirect(adjudicationUrls.hearingDetails.urls.review(adjudicationNumber))
+    } catch (postError) {
+      if (this.pageOptions.isEdit()) {
+        logger.error(`Failed to amend hearing: ${postError}`)
+        res.locals.redirectUrl = adjudicationUrls.scheduleHearing.urls.edit(adjudicationNumber, hearingId)
+      } else {
+        logger.error(`Failed to schedule hearing: ${postError}`)
+        res.locals.redirectUrl = adjudicationUrls.scheduleHearing.urls.start(adjudicationNumber)
+      }
+      throw postError
+    }
+  }
+
+  readFromApi = async (adjudicationNumber: number, hearingId: number, user: User): Promise<HearingDetails> => {
+    const adjudication = await this.reportedAdjudicationsService.getReportedAdjudicationDetails(
+      adjudicationNumber,
+      user
+    )
+    const { reportedAdjudication } = adjudication
+    const [hearingToRender] = reportedAdjudication.hearings.filter(hearing => hearing.id === hearingId)
+    return {
+      hearingDate: await convertDateTimeToObject(hearingToRender.dateTimeOfHearing),
+      locationId: hearingToRender.locationId,
+      id: hearingToRender.id,
+    }
+  }
+
+  getPageDataOnGet = async (
+    adjudicationNumber: number,
+    hearingDetails: HearingDetails,
+    user: User
+  ): Promise<PageData> => {
+    return {
+      displayData: await this.getDisplayData(adjudicationNumber, user),
+      formData: {
+        hearingDetails,
+      },
+    }
+  }
+
+  getDisplayData = async (adjudicationNumber: number, user: User): Promise<DisplayDataFromApis> => {
+    const prisoner = await this.reportedAdjudicationsService.getPrisonerDetailsFromAdjNumber(adjudicationNumber, user)
+    const { agencyId } = prisoner.assignedLivingUnit
+    const possibleLocations = await this.locationService.getIncidentLocations(agencyId, user)
+    return {
+      possibleLocations,
+      adjudicationNumber,
+    }
+  }
+}
+
+const renderData = (res: Response, pageData: PageData, error: FormError) => {
+  const data = {
+    hearingDate: getHearingDate(pageData.formData.hearingDetails?.hearingDate),
+    locationId: pageData.formData.hearingDetails?.locationId,
+  }
+
+  return res.render(`pages/adjudicationTabbedParent/scheduleHearing`, {
+    errors: error ? [error] : [],
+    cancelHref: adjudicationUrls.hearingDetails.urls.review(pageData.displayData.adjudicationNumber),
+    locations: pageData.displayData.possibleLocations,
+    data,
+  })
+}
+
+const extractValuesFromPost = (req: Request): SubmittedFormData => {
+  const values = {
+    hearingDetails: {
+      hearingDate: req.body.hearingDate,
+      locationId: req.body.locationId,
+    },
+  }
+  return values
+}
+
+const getHearingDate = (userProvidedValue?: SubmittedDateTime) => {
```

Comment:
i noticed this is also in incidentDetails page.  Would it better to move all instances of this to a utility class / function?